modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/BatchExportImportPerformanceTest.java

```diff
@@ -0,0 +1,395 @@
+			).put(
+				"twitter", ""
+			).put(
+				"webUrls", JSONFactoryUtil.createJSONArray()
+			)
+		).toString();
+	}
+
+	private static void _writeToLogFile(String... contents) throws IOException {
+		Files.write(
+			_logFilePath, Arrays.asList(contents), StandardOpenOption.APPEND,
+			StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+	}
+
+	private String _createBatchJSON(String className, int recordsCount) {
+		StringBundler batchJsonSB = new StringBundler();
+
+		batchJsonSB.append(StringPool.OPEN_BRACKET);
+
+		for (int i = 0; i < recordsCount; i++) {
+			String alternateName = RandomTestUtil.randomString(
+				8, UniqueStringRandomizerBumper.INSTANCE);
+
+			String json = StringUtil.replace(
+				_jsonTemplates.get(className), "[$ALTERNATE_NAME$]",
+				alternateName);
+
+			json = StringUtil.replace(
+				json, "[$FAMILY_NAME$]",
+				StringUtil.getTitleCase(
+					RandomTestUtil.randomString(
+						8, UniqueStringRandomizerBumper.INSTANCE),
+					true, ""));
+
+			json = StringUtil.replace(
+				json, "[$GIVEN_NAME$]",
+				StringUtil.getTitleCase(
+					RandomTestUtil.randomString(
+						8, UniqueStringRandomizerBumper.INSTANCE),
+					true, ""));
+
+			batchJsonSB.append(
+				StringUtil.replace(
+					json, "[$EMAIL_ADDRESS$]",
+					StringBundler.concat(
+						alternateName, "@", RandomTestUtil.randomString(),
+						".com")));
+
+			if (i < (recordsCount - 1)) {
+				batchJsonSB.append(StringPool.COMMA);
+			}
+		}
+
+		batchJsonSB.append(StringPool.CLOSE_BRACKET);
+
+		return batchJsonSB.toString();
+	}
+
+	private Map<String, String> _splitClassName(String className) {
+		Map<String, String> classNamePartsMap = new HashMap<>();
+
+		if (className.contains("#")) {
+			String[] classNameParts = className.split("#");
+
+			classNamePartsMap.put("className", classNameParts[0]);
+			classNamePartsMap.put("taskItemDelegateName", classNameParts[1]);
+		}
+		else {
+			classNamePartsMap.put("className", className);
+		}
+
+		return classNamePartsMap;
+	}
+
+	private Closeable _startTimer() {
+		Thread thread = Thread.currentThread();
+
+		StackTraceElement stackTraceElement = thread.getStackTrace()[2];
+
+		String invokerName = StringBundler.concat(
+			stackTraceElement.getClassName(), StringPool.POUND,
+			stackTraceElement.getMethodName());
+
+		long startTime = System.currentTimeMillis();
+
+		return () -> {
+			long totalTimeMillis = System.currentTimeMillis() - startTime;
+
+			double speed =
+				(double)(_recordsCount * 1000) / (double)totalTimeMillis;
+
+			_writeToLogFile(
+				StringBundler.concat(
+					invokerName, " used ", totalTimeMillis, " ms, for ",
+					_recordsCount, " records, speed: ",
+					String.format("%.3f", speed), " records/s"));
+		};
+	}
+
+	private void _testPostExportTask(String className) throws Exception {
+		Map<String, String> classNamePartsMap = _splitClassName(className);
+
+		try (Closeable closeable = _startTimer()) {
+			ExportTaskResource.Builder builder = ExportTaskResource.builder();
```

Comment:
If we use the builder we are not testing the whole end to end that the user would experience. If you have a look at [this test](https://github.com/liferay/liferay-portal/blob/12b63d8c3ea4aae2b2c1eac1c99c298ef0485db2/modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/UserAccountResourcePerformanceTest.java#L266-L277) they are using a complete http request which will be much more real. 

It depends on what we really want to test, but I think we want to go for full http.

Reply:
See https://github.com/liferay-headless/liferay-portal/pull/1884/commits/3d617a6cbb9422c9a93633601230c493d3bf135e.